src/cli/compiler.js

```diff
@@ -66,6 +67,80 @@ class Compiler {
     }
   }
 
+  getFileLocations() {
+    try {
+      let allFiles = []
+      let subgraph = this.loadSubgraph()
+
+      // Add all file paths specified in manifest
+      allFiles.push(subgraph.getIn(['schema', 'file']))
+      let mappingFiles = subgraph.get('dataSources').map(dataSource => {
+          allFiles.push(dataSource.getIn(['mapping', 'file']))
+          dataSource.getIn(['mapping', 'abis']).map(abi => {
+            allFiles.push(abi.get('file'))
+          })
+      })
+
+      // Convert all paths to absolute reference
+      let allAbsolutePaths = allFiles.map(file => {
+        return path.resolve(this.sourceDir, file)
+      })
+      return allAbsolutePaths
+    } catch(e) {
+      this.logger.fatal('Failed to parse subgraph file locations:', e)
+    }
+  }
+
+  watchAndCompile() {
+    let compiler = this
+    compiler.logger.info('')
+
+    // Initialize watcher
+    let watcher = chokidar.watch(path.resolve(this.sourceDir, this.options.subgraphManifest), {
+      persistent: true,
+      ignoreInitial: true,
+      ignored: [
+        './dist',
+        './node_modules',
+        /(^|[\/\\])\../
+      ],
+      depth: 3,
+      atomic: 500
+    })
```

Comment:
Since we're only watching the files we're interested in now, some of the above parameters can probably be dropped, e.g. `ignored` and `depth`.